eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -686,9 +686,11 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
             // we were waiting for our pending htlcs to be signed before replying with our local shutdown
             val finalScriptPubKey = getOrGenerateFinalScriptPubKey(d)
             val localShutdown = Shutdown(d.channelId, finalScriptPubKey)
+            // this should always be defined, we provide a fallback for backward compat with older channels
+            val closeStatus = d.closeStatus_opt.getOrElse(CloseStatus.Initiator(None))
```

Comment:
This is needed because is e.g. `ChannelCodecs3.closeStatusCompatCodec`, we will not set the `CloseStatus` if there are no `closingFeerates`. In theory we could compute it based on the `shutdown`s (like in `ChannelTypes4.scala`, but it doesn't seem worth the hassle (we don't use this info anyway).